new/reclassify_amazon_linux_platform_family.md

```diff
@@ -0,0 +1,30 @@
+---
+RFC: unassigned
+Author: Sean OMeara <sean@chef.io>
+Status: Draft
+Type: Standards
+---
+
+# Title
+
+Reclassify Amazon Linux platform_family
+
+## Motivation
+
+   As time passes, Amazon Linux looks less and less like a RHEL clone.
+   Release version numbers do not match EL5, EL6, EL7.
+   Amazon is now a rolling release
+   Different package sets are available
+   Package names and contents are different
+   Service providers are different
+
+## Specification
+
+Change Amazon Linux platform_family from 'rhel' to 'amazon'
```

Comment:
Amazon Linux isn't Fedora derived anymore.

It really truly is its own thing.

The "things" you case-statement around are not compatible with Fedora.

It's a rolling release.

Package wise, Amazon puts semantically meaningful information into package
names (much like Debian),
The package dependency graph deviated a very long time ago.

A Fedora package repo won't work on Amazon, and an Amazon package repo
doesn't work on Fedora.

Service wise, Amazon currently uses sysvinit. Fedora does uses systemd.

Amazon still uses Yum, Fedora use DNF now, etc.

It should be 'amazon/amazon'



On Sat, Jul 11, 2015 at 6:49 PM, Lamont Granquist <notifications@github.com>
wrote:

> In new/reclassify_amazon_linux_platform_family.md
> <https://github.com/chef/chef-rfc/pull/109#discussion_r34417347>:
>
> > +# Title
> > +
> > +Reclassify Amazon Linux platform_family
> > +
> > +## Motivation
> > +
> > +   As time passes, Amazon Linux looks less and less like a RHEL clone.
> > +   Release version numbers do not match EL5, EL6, EL7.
> > +   Amazon is now a rolling release
> > +   Different package sets are available
> > +   Package names and contents are different
> > +   Service providers are different
> > +
> > +## Specification
> > +
> > +Change Amazon Linux platform_family from 'rhel' to 'amazon'
>
> So I think we should change the platform_family here to 'fedora' rather
> than 'amazon'
>
> Having platform_family and platform of fedora/fedora and amazon/amazon
> doesn't allow leveraging any existing code, and we have to update all the
> if cases. If we reuse 'fedora' then code like this just works:
>
> case platform_familywhen 'rhel', 'fedora'
>   # amazon also works fine hereend
>
> For code that needs to specifically look at fedora as a distro and cares
> about things like platform_version then this code will still work:
>
> case platformwhen 'fedora'
>   # inappropriate to match amazon here
>   platform_family >= 21 ? "systemd" : "sysvinit"end
>
> There might be some (bad) code which matches on platform_family of
> 'fedora' and then does very fedora-specific code which would break on
> 'amazon', but that code should be changed to use platform of 'fedora'
> instead of 'platform_family'.
>
> This gets us back to the original design goal of platform_family which is
> to let you be lazy with case statements and have good branching logic. To
> that goal it makes more sense to only have the two 'family' buckets where
> 'rhel' is explicitly 'rebranded redhat enterprise linux with compatible
> platform_version numbers' and 'fedora' is 'all other fedora-derived
> distributions'.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/chef/chef-rfc/pull/109/files#r34417347>.
>